invenio_communities/communities/service.py

```diff
@@ -10,10 +10,72 @@
 
 """Invenio Communities Service API."""
 
+from elasticsearch_dsl import Q
+from invenio_records_resources.services.base import LinksTemplate
+from invenio_db import db
 from invenio_records_resources.services.records import RecordService
+from marshmallow.exceptions import ValidationError
 
 
 class CommunityService(RecordService):
     """community Service."""
 
-    pass
+    def search_user_communities(
+            self, identity, params=None, es_preference=None, **kwargs):
+        """Search for records matching the querystring."""
+        self.require_permission(identity, 'search_user_communities')
+
+        # Prepare and execute the search
+        params = params or {}
+        search_result = self._search(
+            'search',
+            identity,
+            params,
+            es_preference,
+            extra_filter=Q(
+                "term",
+                **{"access.owned_by.user": identity.id}
+            ),
+            permission_action='read',
+            **kwargs).execute()
+
+        return self.result_list(
+            self,
+            identity,
+            search_result,
+            params,
+            links_tpl=LinksTemplate(self.config.links_user_search, context={
+                "args": params
+            }),
+            links_item_tpl=self.links_item_tpl,
+        )
+
+    def rename(self, id_, identity, data, revision_id=None):
+        """Rename a community."""
+        record = self.record_cls.pid.resolve(id_)
+
+        self.check_revision_id(record, revision_id)
+
+        # Permissions
+        self.require_permission(identity, "rename", record=record)
```

Comment:
Missed this one... We need a `can_rename` permission for this to work... Let's keep at a separate permission since in the future when we have multiple community members it will be only the community admin/owner that can rename or delete the community.